extensions/vertx/runtime/src/main/java/io/quarkus/vertx/core/runtime/VertxTimerAwareScheduledExecutorService.java

```diff
@@ -0,0 +1,199 @@
+package io.quarkus.vertx.core.runtime;
+
+import java.util.Collection;
+import java.util.List;
+import java.util.concurrent.Callable;
+import java.util.concurrent.Delayed;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.Future;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+import io.vertx.core.Context;
+import io.vertx.core.Handler;
+import io.vertx.core.Vertx;
+
+/**
+ * A wrapper for {@link ScheduledExecutorService} that dispatches some of the operations using Vert.x timers
+ * when called from Vert.x event-loops, else it just dispatches to a delegate {@link ScheduledExecutorService}.
+ *
+ * This class is used to keep some Mutiny scheduled operations on a Vert.x event-loop thread rather than hop
+ * to a worker thread: delaying items, retries on failures, streams of periodic ticks, etc.
+ */
+public final class VertxTimerAwareScheduledExecutorService implements ScheduledExecutorService {
```

Comment:
In Quarkus core Utils we've `ForwardingExecutorService` and `ForwardingScheduledExecutorService` for the delegation.

Reply:
Good catch, I didn't knew about this one!